### PR TITLE
Graduate course catalog

### PR DESCRIPTION
**Pull request by kabel**
Adding a new controller to handle new routes for graduate optimized course catalog.

### CODE REVIEW

**Comment by mfairchild365** [comment]
This looks good, I went though the code, deployed a test instance and everything appeared to work fine. I'd however like to see some tests associated with this this PR.

File: src/SubjectArea/SubjectArea.php [hunk 1]
```diff
@@ -111,6 +124,43 @@ public function getCourses()
         return $this->courses;
     }
 
+    protected function getGradSortedCourses($courses)
```
Comment (mfairchild365) [inline src/SubjectArea/SubjectArea.php]: a test for this might be helpful

File: src/SubjectArea/FilterWithCourses.php [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+namespace UNL\UndergraduateBulletin\SubjectArea;
+
+use UNL\UndergraduateBulletin\Controller;
+
+class FilterWithCourses extends \FilterIterator
+{
+	protected $controller;
+
+	public function __construct(SubjectAreas $subjects, Controller $controller)
```
Comment (mfairchild365) [inline src/SubjectArea/FilterWithCourses.php]: tabs!

**Comment by kabel** [comment]
There, a fancy test for you @mfairchild365. Could be better but I didn't want to deal with data mocking right now.

File: test/SubjectArea/GraduateSortedCoursesTest.php [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        ];
+
+        $expectedSortedCourses = [
+            $unsortedCourses[1],
+            $unsortedCourses[2],
+            $unsortedCourses[0],
+        ];
+
+        $courses = new GraduateSortedCourses(new \ArrayIterator($unsortedCourses));
+        $courses = iterator_to_array($courses, false);
```
Comment (mfairchild365) [inline test/SubjectArea/GraduateSortedCoursesTest.php]: TIL `iterator_to_array()` is a thing

Comment (kabel) [inline test/SubjectArea/GraduateSortedCoursesTest.php]: For all your iterator to array needs.

**Comment by mfairchild365** [comment]
HOT DOG!!

**Pull request merged**